pkg/pool-stable/contracts/meta/MetaStablePool.sol

```diff
@@ -420,17 +454,53 @@ contract MetaStablePool is StablePool, StableOracleMath, PoolPriceOracle, IPrice
     }
 
     /**
-     * @dev Overrides scaling factor getter to introduce the tokens' price rate
+     * @dev Overrides scaling factor getter to introduce the tokens' price rate.
+     * Note that it may update the price rate cache if necessary.
      */
     function _scalingFactors() internal view virtual override returns (uint256[] memory scalingFactors) {
+        // There is no need to check the arrays length since both are based on `_getTotalTokens`
+        // Given there is no generic direction for this rounding, it simply follows the same strategy as the BasePool.
         scalingFactors = super._scalingFactors();
         uint256[] memory priceRates = _priceRates();
+        scalingFactors[0] = scalingFactors[0].mulDown(priceRates[0]);
+        scalingFactors[1] = scalingFactors[1].mulDown(priceRates[1]);
```

Comment:
Extremely minor, but given we know it's only two rates, I think we can remove `_priceRates` and just call `_priceRate(token0)` here, removing the temporary arrays.